pkg/api/controller_test.go

```diff
@@ -3713,3 +3713,67 @@ func TestController_PostStatsEvents(t *testing.T) {
 		})
 	}
 }
+
+func TestController_CopyObjectHandler(t *testing.T) {
+	clt, deps := setupClientWithAdmin(t)
+	ctx := context.Background()
+
+	_, err := deps.catalog.CreateRepository(ctx, "repo1", onBlock(deps, "bucket/prefix"), "main")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	const expectedSize = 38
+	const content = "hello world this is my awesome content"
+	uploadResp, err := uploadObjectHelper(t, ctx, clt, "foo/bar", strings.NewReader(content), "repo1", "main")
+	verifyResponseOK(t, uploadResp, err)
+
+	sizeBytes := api.Int64Value(uploadResp.JSON201.SizeBytes)
+
+	if sizeBytes != expectedSize {
+		t.Fatalf("expected %d bytes to be written, got back %d", expectedSize, sizeBytes)
+	}
+
+	t.Run("copy object", func(t *testing.T) {
```

Comment:
suggestion: Test on the same/different branch - they both copy now, but soon that won't be the case :)

Reply:
When we will support cross branch copy - we will add the relevant tests